### ISSUE TITLE
ArrayBuffer constructor mismatch, instanceof with the global doesn't work

### ISSUE THREAD

**Issue by panva** [vercel/edge-runtime#62]
```
cat node_modules/edge-runtime/package.json | jq '.version'
"1.1.0-beta.22"
```

```
const ab = await crypto.subtle.digest('SHA-256', crypto.getRandomValues(new Uint8Array(32)))
console.log(ab instanceof ArrayBuffer)
console.log(ab.constructor)
```

Despite `ab.constructor` showing an `ArrayBuffer`, `ab instanceof ArrayBuffer` is false.

**Comment by panva** [comment]
FYI I only do the following to get these bugs to surface.

```
gh repo clone panva/oauth4webapi
cd oauth4webapi
npm clean-install
npm install edge-runtime@latest
npm run tap:edge-runtime
```

**Comment by Schniz** [comment]
Hey @panva , this is fixed in the main branch right now. Unfortunately, the REPL will not show it as fixed as it is using the host ArrayBuffer instead of the vm's ArrayBuffer, but with the fixes from #80. I opened that PR because I tried to verify that my fix works for short test case and it did not fix it, then I realized that the issue is in the REPL itself, so I wanted to make sure you know that :pray:
